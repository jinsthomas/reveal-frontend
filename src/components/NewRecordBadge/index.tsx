import moment from 'moment';
import React from 'react';
import { Badge } from 'reactstrap';

/** interface for NewRecordBadge props */
interface NewRecordBadgeProps {
  className: string;
  numDays: number;
  recordDate: string | Date | null;
}

/** Simple component that displays a badge for new records */
const NewRecordBadge = (props: NewRecordBadgeProps) => {
  const { className, numDays, recordDate } = props;
  if (
    recordDate &&
    moment(recordDate) &&
    moment(recordDate).isAfter(moment().subtract(numDays, 'days'))
  ) {
    return (
      <Badge color="warning" pill={true} className={className}>
        New
      </Badge>
    );
  }
  return null;
};

/** define default props */
const defaultProps: NewRecordBadgeProps = {
  className: 'new-record',
  numDays: 1,
  recordDate: null,
};

/** declare default props for NewRecordBadge */
NewRecordBadge.defaultProps = defaultProps;

export default NewRecordBadge;
