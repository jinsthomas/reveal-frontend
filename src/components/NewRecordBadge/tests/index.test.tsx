import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewRecordBadge from '..';

describe('components/NewRecordBadge', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(<NewRecordBadge />);
  });
});
