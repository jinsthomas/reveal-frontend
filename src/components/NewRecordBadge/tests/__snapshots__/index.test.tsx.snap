// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NewRecordBadge renders badge correctly: badge 1`] = `
<Badge
  className="new-record"
  color="warning"
  pill={true}
  tag="span"
>
  <span
    className="new-record badge badge-warning badge-pill"
  >
    New
  </span>
</Badge>
`;
